processing/src/main/java/org/apache/carbondata/processing/loading/parser/CarbonParserFactory.java

```diff
@@ -51,23 +54,37 @@ public static GenericParser createParser(CarbonColumn carbonColumn, String[] com
    *                           delimiters
    * @return GenericParser
    */
-  private static GenericParser createParser(CarbonColumn carbonColumn, String[] complexDelimiters,
+  private static GenericParser createParser(CarbonColumn carbonColumn,
+      Queue<String> complexDelimiters,
       String nullFormat, int depth) {
+    if (depth > 2) {
+      return null;
```

Comment:
I think we should throw exception with proper error message if depth is more than 2

Reply:
done